platform/pulpcore/app/tasks/base.py

```diff
@@ -0,0 +1,56 @@
+from celery import shared_task
+from django.http import QueryDict
+
+from pulpcore.tasking.tasks import UserFacingTask
+from pulpcore.app.apps import get_plugin_config
+
+
+@shared_task(base=UserFacingTask)
+def general_update(instance_id, app_label, serializer_name, *args, **kwargs):
+    """
+    Update a model
+
+    The model instance is identified using the app_label, id, and serializer name. The serializer is
+    used to perform validation.
+
+    Args:
+        id (str): the id of the model
+        app_label (str): the Django app label of the plugin that provides the importer
+        serializer_name (str): name of the serializer class for the importer
+        data (dict): dictionary whose keys represent the fields of the importer that need to be
+            updated with the corresponding values.
+        partial (bool): When true, only the fields specified in the data dictionary are updated.
+            When false, any fields missing from the data dictionary are assumed to be None and
+            their values are updated as such.
+
+    Raises:
+        :class:`rest_framework.exceptions.ValidationError`: When serializer instance can't be saved
+            due to validation error. This theoretically should never occur since validation is
+            performed before the task is dispatched.
+    """
+    data = kwargs.pop('data', None)
+    partial = kwargs.pop('partial', False)
+    serializer_class = get_plugin_config(app_label).named_serializers[serializer_name]
+    instance = serializer_class.Meta.model.objects.get(id=instance_id).cast()
+    data_querydict = QueryDict('', mutable=True)
+    data_querydict.update(data)
```

Comment:
I don't think its super important, but it might be a good idea to set the QueryDict to immutable after the update.